File: SocialWebApp/Domain/Entities/Message.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entities;
public class Message
{
  public int Id { get; set; }
  public string Content { get; set; }
  public int Type { get; set; }
  public DateTime CreatedAt { get; set; }
  public bool IsRead { get; set; }

  public int SenderId { get; set; }
  public User? Sender { get; set; }

  public int ReceiverId { get; set; }
  public User? Receiver { get; set; }

  public bool IsDeleted { get; set; } = false;
}

File: SocialWebApp/WebUI/ClientApp/src/app/interface/message.ts
import { IUser, IUserHub } from './user';

export interface IMessage {
  id?: number;
  content: string;
  type: MessageContentType;
  createdAt: Date;
  isRead: boolean;
  senderId: number;
  receiverId: number;
  sender?: IUser;
  receiver?: IUser;
}
export interface IMessageHub {
  Id?: number;
  Content: string;
  Type: MessageContentType;
  CreatedAt: Date;
  IsRead: boolean;
  SenderId: number;
  ReceiverId: number;
  Sender: IUserHub;
  Receiver: IUserHub;
}

export enum MessageContentType {
  TEXT,
  IMAGE,
  DOC
}

export enum MessageType {
  SEND,
  RECEIVE
}
